File: Miscallaneous/codeforces/1328B.py
def solve():
    n_k = input().split(' ')
    n = int(n_k[0])
    k = int(n_k[1])
    b = 1
    while k>0:
        k -= b
        b += 1
    k = -k
    b -= 1
    for i in range(1, n+1):
        if i==n-b or i==n-b+k+1:
            print('b', end = '')
        else:
            print('a', end = '')
    print('')

t = int(input())
for i in range(0, t):
    solve()

File: Miscallaneous/codeforces/1328B.c
#include <stdio.h>

void solve(void);

int main(void)
{
    int i, t;
    scanf("%d", &t);
    for(i=0; i<t; i++)
        solve();
    
    return 0;
}

void solve(void)
{
    int n, k, b;
    scanf("%d %d", &n, &k);
    for(b=1; k>0; b++)
        k -= b;
    k = -k; b--;

    int i;
    for(i=1; i<=n; i++)
    {
        if(i==n-b || i==n-b+k+1)
            printf("b");
        else
            printf("a");
    }
    printf("\n");
}